subprojects/logging/src/main/java/org/gradle/util/SingleMessageLogger.java

```diff
@@ -41,7 +42,8 @@ protected Boolean initialValue() {
 
     public static final String INCUBATION_MESSAGE = "%s is an incubating feature.";
 
-    private static LoggingDeprecatedFeatureHandler deprecatedFeatureHandler = new LoggingDeprecatedFeatureHandler();
+    @VisibleForTesting
+    static LoggingDeprecatedFeatureHandler deprecatedFeatureHandler = new LoggingDeprecatedFeatureHandler();
```

Comment:
Replacing the static field in a unit test doesn't seem right. This static field should be private and final.

If you have some idea how to make this better please go ahead. If not, changing the visibility here at least shows that we have a problem with the unit test.

Reply:
OK, reverted this.

Reply:
I was actually thinking that the version annotated with `@VisibleForTesting` was better, since it answers the question why the field is not private. And it shows in the production code that there may be a problem with the test.

Anyway, feel free to leave it in the way you think is best.